src/react-components/room/ReactionButtonContainer.js

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState, useCallback } from "react";
+import PropTypes from "prop-types";
+import { ReactComponent as ReactionIcon } from "../icons/Reaction.svg";
+import { ToolbarButton } from "../input/ToolbarButton";
+
+export function ReactionButtonContainer({ scene }) {
+  const [isFrozen, setIsFrozen] = useState(scene.is("frozen"));
+
+  useEffect(
+    () => {
+      function onSceneStateChange(event) {
+        if (event.detail === "frozen") {
+          setIsFrozen(scene.is("frozen"));
+        }
+      }
+
+      scene.addEventListener("stateadded", onSceneStateChange);
+      scene.addEventListener("stateremoved", onSceneStateChange);
+
+      return () => {
+        scene.removeEventListener("stateadded", onSceneStateChange);
+        scene.removeEventListener("stateremoved", onSceneStateChange);
+      };
+    },
+    [scene]
+  );
+
+  // TODO: We probably shouldn't use freeze mode for users spawning emojis on a 2d device.
```

Comment:
+1 - Good idea to either implement a 2D emoji menu or separate the logic that ties their visibility to the frozen state. https://github.com/mozilla/hubs/blob/7159d67ccc84d6dfccf7bf3b1590275f04092367/src/components/emoji-hud.js#L88-L91

Reply:
I like the suggested design of the emoji popover menu as a first (or second) pass. I think eventually we'll want something integrated into the chat sidebar and easily extended / searchable like slack or discord's emoji menu:
![image](https://user-images.githubusercontent.com/4072106/97254689-26ffd600-17cc-11eb-9828-d0e665507bcf.png)
![image](https://user-images.githubusercontent.com/4072106/97254739-4565d180-17cc-11eb-996c-1098dd85a496.png)

Reply:
I especially like that when Discord's emoji button is hovered, it will change the emoji each time. I would like to do the same for our emoji button. Maybe it would fit for our "Place" menu too.